articles/osquery-evented-tables-overview.md

```diff
@@ -0,0 +1,161 @@
+- [What are osquery evented tables?](#what-are-osquery-evented-tables?)
+- [How do evented tables work?](#how-do-evented-tables-work?)
+- [What do I need to consider when configuring evented tables?](#what-do-i-need-to-consider-when-configuring-evented-tables?)
+- [How do I turn on an evented table?](#how-do-i-turn-on-an-evented-table?)
+-- [File integrity monitoring](#file-integrity-monitoring-fim)
+-- [Process auditing](#process-auditing)
+-- [Yara scanning](#yara-scanning)
+
+# Osquery evented tables
+
+## What are osquery evented tables? 
+Osquery traditionally allows you to see the state of your computers right now. This point-in-time snapshot, however, requires you to actively watch the table repeatedly. For example, in the security-critical `processes` table, a process might be very short-lived and there is high churn in processes. Actively watching and diffing this table would be challenging even with automation.
+
+This is where osquery's evented tables come in. Rather than displaying the point-in-time state for your host, osquery's evented tables store the host's historical data. You can configure osquery to capture certain types of information, after which it will be stored in the relevant `*_events` table for later analysis. 
+
+This guide will walk you through the concepts, considerations, and best practices for setting up evented tables, in particular focusing on osquery. It will also provide basic information about commonly-used evented tables for a newcomer to get started.
+
+## How do osquery evented tables work?
+Osquery does not generate the events itself. Instead, it reads and formats event data generated by system utilities. For example, on Linux, audit generates process and socket event data in its own storage location and format. Osquery reads data from audit’s storage location, converts it into an event row, and buffers them (in the internal [rocksdb](http://rocksdb.org) store) in the `process_events` and `socket_events` tables to await querying. The data can then be filtered and transformed via SQL, and shipped to a log destination with the scheduled query functionality. 
+
+For the purposes of this article, we will use the term "utility" to mean the underlying OS utility that osquery subscribes to for its various evented tables. 
```

Comment:
> we will use the term "utility" to mean the underlying OS utility

I still think it is cleaner to have `underlying OS APIs/functionality` and not use the word utility there

Reply:
Quite a mouthful, I think as long as we define our terms up front, we'd be ok. What's wrong with utility?

Reply:
> Quite a mouthful 

I disagree on this :) How is that a mouthful? `underlying OS utility` --> `underlying OS APIs` or `underlying OS components`? 

> What's wrong with utility?

Utility in this context usually means that it's a program/binary on the system. This would imply that we are shelling out to that binary, which is not the case. We are using the same underlying APIs that that program would use.

Reply:
How about

> the underlying OS capabilities

Avoids all the baggage that widespread webdev has clouded the words "API" and "components" with (has semantic competition with ideas like REST APIs and UI components).

Reply:
I've updated the first few lines to be more explicit that it is using an underlying OS component, not an OS utility. 

I still think defining the term utility up front is sufficient. And I'm not sure our intended audience will have the distinction between shelling vs. the underlying API in mind, or if they did, they would mind this distinction between elided. 

I would like to ship this and if our intended audience doesn't get it due to this distinction, then we can fix. Our original requester can read it and tell us. I also have sent it to someone else who expressed interest in this article.

Reply:
Sounds good to me, thanks @zhumo